tdh/resource_user.go

```diff
@@ -348,19 +408,60 @@ func (r *userResource) Read(ctx context.Context, req resource.ReadRequest, resp
 		return
 	}
 
-	// Get refreshed cluster value from TDH
-	user, err := r.client.CustomerMetadata.GetUser(state.ID.ValueString())
-	if err != nil {
-		resp.Diagnostics.AddError(
-			"Reading TDH user",
-			"Could not read TDH user ID "+state.ID.ValueString()+": "+err.Error(),
-		)
-		return
-	}
+	query := &customer_metadata.UsersQuery{}
+	if r.client.Root.IsSre {
+		user, err := r.client.CustomerMetadata.GetUsers(query, r.client.Root.IsSre)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Reading TDH user",
+				"Could not read TDH user ID "+state.ID.ValueString()+": "+err.Error(),
+			)
+			return
+		}
```

Comment:
you are fetching all the users to match single id??

Reply:
is there no API like `/users/<ID>` for SRE ?
or ID query param on existing `/users` API ?

Reply:
there is no api for this

Reply:
existing api is not working in are